plugins/woocommerce-blocks/bin/webpack-configs.js

```diff
@@ -406,7 +406,8 @@ const getFrontConfig = ( options = {} ) => {
 						name: 'wc-blocks-frontend-vendors',
 						chunks: ( chunk ) => {
 							return (
-								chunk.name !== 'product-button-interactivity'
+								chunk.name !== 'product-button-interactivity' ||
+								chunk.name !== 'product-image-interactivity'
```

Comment:
Hmm I am curious about this change. I am not super familiar with Webpack. Can you please explain why we need to add the `product-image-interactivity` entry here? 🙂

Reply:
Hmm... that's a good question and I must confess I followed changes for `product-button-interactivity` and I agree, not the best approach 😅 

@samueljseay , sorry for ping out of nowhere. Before I waste lots of time trying to understand it, do you remember by any chance what was the reason to exclude `product-button-interactivity` chunk here? Does it make sense to do the same for Product Image frontend interactivity chunk?

Reply:
Product button loads a lot of dependencies unfortunately that are not shared across all bundles so it was inflating the size of this vendors file which is here to optimize performance by combining dependencies needed by many entry points within this bundle.

Answering whether this should be here is difficult to say without using the webpack bundle analyzer.

It might be faster for you to validate yourself. To do so you can run the bundle analyzer with and without this change and pay special attention to:

* overall bundle size  (this is the frontend bundle)
* check if this `wc-blocks-frontend-vendors` script increases in size dramatically

It's basically a trade off of the above to answer whether this is worth it. 

For example:

lets say product-image-interactivity and its dependencies is 10KB and by not optimizing it we add 10KB overall to the whole bundle.

but lets say 7kb of product-image-interactivity is dependencies already available on the vendors bundle, then optimizing here would only add 3kb to the whole bundle then its probably worth doing.

Reply:
Thanks @samueljseay for the explanation, appreciate it!

I'd say the results are somewhat surprising (all gzipped):

Bundle | Exclude `'product-image-interactivity'` from vendors | Include `'product-image-interactivity'` in vendors
-- | -- | --
All | 155.74 KB | 154.37 KB
`wc-blocks-frontend-vendors-frontend` | 30.3KB | 25.49KB
`product-image-interactivity-frontend` | 655B | 655B

But looking at the results we should include the `product-image-interactivity` in vendors hence I removed the condition in https://github.com/woocommerce/woocommerce/pull/50217/commits/5bc4ca79df8701e8b290284689619277dd3c6625. 🙌 

@imanish003 , do you think it's good to go then?

Reply:
@kmanijak I really appreciate you taking the time to investigate this thoroughly. Looks good to me  🚀

Reply:
weird how the vendor bundle dropped 5KB 😂  🚢